### PR TITLE
Prepares 3.3.7 release

### PR DESCRIPTION
**Pull request by carolabadeer**
*Issue #, if available:* N/A

*Description of changes:*

Similar to #511 

Bump AWS X-Ray Javascript SDK components to version `3.3.7` and update CHANGELOG.md. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by jj22ee — APPROVED** [review]
LGTM!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,30 @@
...
+## 3.3.7
+View [the latest changes](https://github.com/aws/aws-xray-sdk-node/compare/aws-xray-sdk-node%403.3.6...aws-xray-sdk-node%403.3.7)
+
+* Added node 16 testing [PR #513](https://github.com/aws/aws-xray-sdk-node/pull/513)
+* Added opentelemetry JS links to README [PR #514](https://github.com/aws/aws-xray-sdk-node/pull/514)
+
+* `aws-xray-sdk-core` updated to 3.3.7
+  * fix: Stop throwing unnecessary errors with adding annotations, metadata, errors [PR #467](https://github.com/aws/aws-xray-sdk-node/pull/467)
+  * fix: Fix TS declaration of TraceId to match definition [PR #520](https://github.com/aws/aws-xray-sdk-node/pull/520)
+  * fix: Fix prototype pollution issue [PR #529](https://github.com/aws/aws-xray-sdk-node/commit/89769989d46eed9e44d8af743e6fb14ad924d48b)
```
Comment (willarmiros) [inline CHANGELOG.md:14]: Change to PR link

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,30 @@
 # Changelog for AWS X-Ray SDK for JavaScript
-<!--LATEST=3.3.6-->
+<!--LATEST=3.3.7-->
 <!--ENTRYINSERT-->
+
+## 3.3.7
+View [the latest changes](https://github.com/aws/aws-xray-sdk-node/compare/aws-xray-sdk-node%403.3.6...aws-xray-sdk-node%403.3.7)
+
+* Added node 16 testing [PR #513](https://github.com/aws/aws-xray-sdk-node/pull/513)
+* Added opentelemetry JS links to README [PR #514](https://github.com/aws/aws-xray-sdk-node/pull/514)
```
Comment (willarmiros) [inline CHANGELOG.md:9]: Do not include these types of "internal" PRs in the changelog - we only include changes that impact behavior of SDK

**Review by willarmiros — APPROVED** [review]
lgtm after addressing comments

**Review by willarmiros — APPROVED** [review]

**Pull request merged**